Automattic/jetpack
# Whitelist (allow) jetpack_sync_settings_* options to be synced

As part of ongoing work to confirm the validity of the Cache Sites we need to ensure that we have a 1-1 match off allow and disallow lists for synced content. This change adds the following options to the whitelist (allowlist):

'jetpack_sync_settings_comment_meta_whitelist',
'jetpack_sync_settings_post_meta_whitelist',
'jetpack_sync_settings_post_types_blacklist',
'jetpack_sync_settings_taxonomies_blacklist',

Please note that it builds off #17928 that re-orders the list. For full end to end testing D53529 needs to be applied to WP.com sandboxes.

#### Changes proposed in this Pull Request:
* Whitelist (allow) jetpack_sync_settings_* to be synced.

#### Jetpack product discussion
Cache Site Integrity is important to ensure all Jetpack Functionality is working as intended for users. This change is to allow automated validation and fixes of discrepancies to the cache site.

#### Does this pull request change what data or activity we track or use?
* No, this data is accessible via API and we actively pull it for several uses. This simply persists so we don't require additional API calls during runtime.

#### Testing instructions:

1) Apply this PR
2) Sandbox the Jetpack connection to your WP.com sandbox, with appropriate http enablement.
3) Apply D53529 to your sandbox.
4) Via the Jetpack Debugger update the "Post type blacklist:" in Sync Settings (add or remove values)
5) On the Remote site execute `Automattic\Jetpack\Sync\Settings::get_blacklisted_post_types_sql();` and verify the added/removed entries are as expected.
6) On the Cache site execute `Automattic\Jetpack\Sync\Settings::get_blacklisted_post_types_sql();` and verify added/removed entries are as expected

#### Proposed changelog entry for your changes:
* Jetpack Sync :: Whitelist sync settings for checksum validation.
Note that this probably doesn't need a changelog as other checksum work will be included so a single entry on that will be sufficient.
